### PR TITLE
Improve machine test coverage

### PR DESCRIPTION
**Pull request by n0npax**
To keep PR small It's improving coverage just for a single file.
I'll provide improvements for remaining files later.

### CODE REVIEW

File: pkg/minikube/machine/cache_binaries_test.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+		t.Fatalf("error during creating tmp dir: %v", err)
+	}
+	defer os.RemoveAll(noWritePermDir)
+	err = os.Chmod(noWritePermDir, 0000)
+	if err != nil {
+		t.Fatalf("error (%v) during changing permissions of dir %v", err, noWritePermDir)
+	}
+	noPermsDir := "/etc"
+	if runtime.GOOS == "windows" {
+		noPermsDir = "C:\\Windows\\System32"
```
Comment (n0npax) [inline pkg/minikube/machine/cache_binaries_test.go]: I don't have access to any windows machine, so this should work on windows, but It wasn't tested.

File: pkg/minikube/machine/cache_binaries_test.go [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+	noWritePermDir, err := ioutil.TempDir("/tmp", "")
+	if err != nil {
+		t.Fatalf("error during creating tmp dir: %v", err)
+	}
+	defer os.RemoveAll(noWritePermDir)
+	err = os.Chmod(noWritePermDir, 0000)
+	if err != nil {
+		t.Fatalf("error (%v) during changing permissions of dir %v", err, noWritePermDir)
+	}
+	noPermsDir := "/etc"
```
Comment (medyagh) [inline pkg/minikube/machine/cache_binaries_test.go]: could you add a comment what is NoPermsDir and why ?

Comment (n0npax) [reply pkg/minikube/machine/cache_binaries_test.go]: comment added (https://github.com/kubernetes/minikube/pull/5512/files#diff-ad5f1899d0f1fcc73de6ef1008d741a9R131)

Comment (tstromberg) [reply pkg/minikube/machine/cache_binaries_test.go]: There is no guarantee that the user running the test does not have write access to /etc.

I would either remove this check, or create another directory using `ioutil.TempDir` and set the permissions appropriately for it.

Comment (n0npax) [reply pkg/minikube/machine/cache_binaries_test.go]: removed

**Comment by n0npax** [comment]
`noPermsDir` removed from test. Thanks for previous review.
Can you please re-review?

**Comment by tstromberg** [comment]
Looks good. Thank you!

**Pull request merged**